pkg/controller/common/defaults/pvc.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package defaults
+
+import corev1 "k8s.io/api/core/v1"
+
+// AppendDefaultPVCs appends defaults PVCs to a set of existing ones.
+//
+// The default PVC is not appended if:
+// - a Volume with the same .Name is found in podSpec.Volumes, and that volume is not a PVC volume
+// - any PVC has been defined by the user
```

Comment:
nit: I feel like it's easier to understand by switching the order of these 2 statements.

```
Defaults are not appended if the user already provided some PVCs, or if the user already provided the default volumes (same name) in the podSpec.
```